### PR TITLE
Products Branch

### PR DESCRIPTION
**Pull request by tejas-muthya37**
This PR is for the Products page of the app which includes a Card Component and the filters functionality.

The code to be reviewed can be found in Screens -> Products -> Products.js

The implementation of the filter functionality can be found in src -> Context -> filter-context.js

### CODE REVIEW

**Review by Ronak-Mutha — APPROVED** [review]

File: src/Screens/Products/Products.js [hunk 1]
```diff
@@ -0,0 +1,355 @@
...
+import Navbar from "./../../Components/Navbar/Navbar";
+import { useParams } from "react-router";
+import { useNavigate } from "react-router-dom";
+
+function Products(props) {
+  let navigate = useNavigate();
+  let { categoryName } = useParams();
+  const inputRef = useRef(null);
+
+  const inputs = document.querySelectorAll('input:not([type="text"])');
```
Comment (shrey27) [inline src/Screens/Products/Products.js]: we should avoid direct DOM manipulation in React JS
Use controlled states for handling inputs

File: src/Components/Navbar/Navbar.js [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+    const items = document.querySelectorAll(".nav-items li");
+    const burger = document.querySelector(".burger");
+    const nav = document.querySelector(".nav-items");
+
+    nav.classList.toggle("navSlide");
+    items.forEach((item) => {
+      item.style.animation = "";
+    });
+    burger.classList.toggle("toggle");
+  };
```
Comment (shrey27) [inline src/Components/Navbar/Navbar.js:30]: Using JS to directly manipulate variables is an anti pattern for React JS
Prefer using controlled states for handling such behaviour

File: src/Components/Navbar/navbar.css [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+    background-color: var(--body-bg);
+    padding: 0 2rem;
+}
+
+.nav-items {
+    display: flex;
+    justify-content: space-between;
+    align-items: center;
+    list-style: none;
+    color: whitesmoke;
```
Comment (shrey27) [inline src/Components/Navbar/navbar.css:31]: use root colors instead of hardcoding them

**Review by shrey27 — CHANGES_REQUESTED** [review]
Please get your code reviewed by TA or MA once before merging
I think there is a lot of scope for improvement

File: src/Components/Navbar/Navbar.js (hunk 2 above)
Comment (tejas-muthya37) [reply src/Components/Navbar/Navbar.js:30]: I have fixed this by using a State hook. Thanks for pointing it out, Shrey.

File: src/Screens/Products/Products.js (hunk 1 above)
Comment (tejas-muthya37) [reply src/Screens/Products/Products.js]: Fixed. Used State hooks instead. Thanks again :)

**Review by SJTGSHIVAM — APPROVED** [review]
As you have done all the required rework I believe you are good to merge this one. Keep in mind dom manipulation is to be avoided in react.

**Comment by shrey27** [comment]
Your rework looks good. No need for further changes.

**Pull request merged**